druid/src/widget/env_scope.rs

```diff
@@ -21,10 +21,16 @@ use tracing::instrument;
 
 /// A widget that accepts a closure to update the environment for its child.
 pub struct EnvScope<T, W> {
-    pub(crate) f: Box<dyn Fn(&mut Env, &T)>,
     pub(crate) child: WidgetPod<T, W>,
+    pub(crate) current_child_env: Option<Env>,
+    pub(crate) prev_super_env: Option<Env>,
+    pub(crate) overrides: Env,
+    pub(crate) modify_env: Option<Box<dyn Fn(&T, &mut Env)>>,
+    pub(crate) should_modify_env_now: EnvInvalidationCheck<T>,
```

Comment:
My understanding is that you can provide *either* an `Env` or a `Fn(&T, &mut Env)`; these are the two supported ways to update the env. Is this correct? what does it mean if the user supplies both of these? Is that fine? I don't see this documented anywhere, which will be important.

Reply:
I was thinking you can set the base environment, but having the closure allows you to dynamically modify only part of the environment you've set or maybe even a different set of variables from the original static set. If it were either dynamic or static then, the dynamic part would have some parts that are static and some parts dynamic, though this wouldn't actually matter practically. 

You're thinking of something like what `Label` does where it has an Enum that holds a static Env or a closure that will be called when data is updated?

Would there ever be a reason to want to change the closure? Say you wanted to update your app some kind of way, but the user changed some setting, so the environment variables you want to change are different. Maybe there should be a way to set the closure like `set_dynamic`?

Reply:
I can't think of good use cases for them being two separate fields versus being in an enum that has `Static` and `Dynamic`.

Reply:
> I was thinking you can set the base environment, but having the closure allows you to dynamically modify only part of the environment you've set or maybe even a different set of variables from the original static set. If it were either dynamic or static then, the dynamic part would have some parts that are static and some parts dynamic, though this wouldn't actually matter practically.
> 
> You're thinking of something like what `Label` does where it has an Enum that holds a static Env or a closure that will be called when data is updated?
> 

This is *exactly* what I was thinking.


> Would there ever be a reason to want to change the closure? Say you wanted to update your app some kind of way, but the user changed some setting, so the environment variables you want to change are different. Maybe there should be a way to set the closure like `set_dynamic`?

I get a sense of YAGNI about this; I think if someone ever actually encounters this problem we can consider it? A closure has a lot of flexibility, you could have various branches in the closure that respond to existing env keys, for instance.

Reply:
Wait, so which part was what you were thinking? Is it fine that the `Env` be set statically and have the option to dynamically update? Or should it be similar in design to how `Label` is constructed?

Reply:
I think something like `Label` would make a lot of sense and be pretty easy to understand.